src/azure-cli/azure/cli/command_modules/monitor/operations/metric_alert.py

```diff
@@ -213,17 +219,35 @@ def _parse_action_removals(actions):
 
 
 def _parse_resource_type(scopes):
-    from msrestazure.tools import parse_resource_id
+    from azure.mgmt.core.tools import parse_resource_id
     from azure.cli.core import CLIError
+
     namespace = None
     resource_type = None
-    for item in scopes:
-        item_namespace = parse_resource_id(item)['namespace']
-        item_resource_type = parse_resource_id(item)['resource_type']
-        if namespace is None and resource_type is None:
+    scope_type = None
+
+    def set_fields(item_namespace, item_resource_type, item_scope_type):
+        nonlocal namespace
+        nonlocal resource_type
+        nonlocal scope_type
+        if namespace is None and resource_type is None and scope_type is None:
             namespace = item_namespace
             resource_type = item_resource_type
+            scope_type = item_scope_type
         else:
-            if namespace != item_namespace or resource_type != item_resource_type:
+            if namespace != item_namespace or resource_type != item_resource_type or scope_type != item_scope_type:
```

Comment:
Can we separate validation logic from main logic?

Reply:
Done